portfolio/src/main/java/com/google/sps/data/BookResponseParser.java

```diff
@@ -0,0 +1,177 @@
+package com.google.sps.data;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.sps.data.Book;
+import com.google.sps.data.BookFieldsEnum;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+// Given a json String from Google Book Api, converts the response into a in-memory Book
+// representation.
+public class BookResponseParser {
+  private static final String VOLUME_INFO = "volumeInfo";
+  private static final String TITLE = "title";
+  private static final String GENRE = "genre";
+  private static final String CATEGORIES = "categories";
+  private static final String AUTHORS = "authors";
+  private static final String LANGUAGE = "language";
+  private static final String DESCRIPTION = "description";
+  private static final String INFO_LINK = "infoLink";
+  private static final String PAGE_COUNT = "pageCount";
+  private static final String PUBLISHED_DATE = "publishedDate";
+  private static final String PUBLISHER = "publisher";
+  private static final String MATURITY_RATING = "maturityRating";
+  private static final String THUMBNAIL = "thumbnail";
+  private static final String IMAGE_LINKS = "imageLinks";
+  private static final String ISBN = "ISBN_13";
+
+  public static Book parseBook(String jsonResponse) {
+    JSONObject jsonObject = new JSONObject(jsonResponse);
+    return jsonToBook(jsonObject);
+  }
+
+  private static Book jsonToBook(JSONObject jsonObject) throws JSONException {
+    String tempTitle;
+    String tempLang;
+    String tempDesc;
+    String tempInfoLink;
+    String tempPubDate;
+    String tempPublisher;
+    String tempMatRate;
+    int tempPageCount;
+    String tempThumbnail; // nested in map {}
+    String tempIsbn; // double nested [{}]
+    ArrayList<String> tempCategories = new ArrayList<String>();
+    ArrayList<String> tempAuthors = new ArrayList<String>();
+
+    Book.Builder builder = Book.builder();
+
+    tempIsbn = getIsbn(jsonObject).get(ISBN);
+    builder.isbn(tempIsbn);
+
+    try {
+      if (jsonObject.getJSONObject(VOLUME_INFO).has(TITLE)) {
```

Comment:
It might be worthwhile to define a local variable to avoid repetition and use it everywhere:

`JSONObject volumeInfo = jsonObject.getJSONObject(VOLUME_INFO);`